source/common/tracing/zipkin/zipkin_core_types.cc

```diff
@@ -71,6 +71,13 @@ Annotation& Annotation::operator=(const Annotation& ann) {
   return *this;
 }
 
+void Annotation::changeEndpointServiceName(const std::string& service_name) {
+  if (endpoint_.valid()) {
+    Endpoint& endpoint = const_cast<Endpoint&>(endpoint_.value());
```

Comment:
using const_cast is generally an indication that there is a design problem. Please refactor so const_cast is not necessary. If you need design help we can discuss the specifics.